rust/kernel/pages.rs

```diff
@@ -25,18 +25,20 @@ pub struct Pages<const ORDER: u32> {
 impl<const ORDER: u32> Pages<ORDER> {
     /// Allocates a new set of contiguous pages.
     pub fn new() -> Result<Self> {
-        // TODO: Consider whether we want to allow callers to specify flags.
+        Self::new_with_flags(bindings::GFP_KERNEL | bindings::__GFP_ZERO | bindings::__GFP_HIGHMEM)
+    }
+
+    /// Allocates a new set of contiguous pages with the provided GFP
+    /// mask (`gfp_mask`).
+    pub fn new_with_flags(gfp_mask: bindings::gfp_t) -> Result<Self> {
```

Comment:
Eventually we want to make the `bindings` module private, so adding it to a public interface is something we should really avoid unless really needed.

Part of the reason I added a `TODO` here instead of doing something like you did was because this requires a little more thought (and something we discussed in today's meeting), for example: do we want to make it just a boolean atomic/sleepable? Or do we want to allow all C behaviours (which means exposing all GFP flags)? Can we infer the atomic/sleepable information from some other context? If we do decide to expose flags, do want to name them exactly the same as the C side? (There are advantages and disadvantages to both choices.)

Reply:
Thanks for the review and sorry for the late reply! 

> Are you blocked on this? If so, and assuming you only need an atomic allocation here.


Yes, I require an atomic allocation.

> I think the simplest thing to do for now is to add a new_atomic variant to unblock you.

That seems a viable solution to the problem, considering that the `bindings` module will be privatized. I will update the commit to add `new_atomic` instead.

> Also, for the Signed-off-by tag, I'm afraid the kernel requires you to use your real name (see https://www.kernel.org/doc/html/latest/process/submitting-patches.html).

Alrighty, I will update the signed-off tag!